Update to make explain plug-in PHP 7 compatible

dunron opened the pull request:
Functions ereg and split have been removed in PHP 7.

dunron committed:
Update to make explain plug-in PHP 7 compatible

splitbrain commented on syntax.php:
```diff
@@ -72,11 +72,11 @@ function _link($target) {
 
         /* Match an internal link. */
-        list($id, $hash) = split('#', $target, 2);
+        list($id, $hash) = preg_split('#', $target, 2);
```
this split should be changed into an explode

dunron replied:
It works with delimiters:

`       list($id, $hash) = preg_split('/#/', $target, 2);`

or with explode:

`       list($id, $hash) = explode('#', $target, 2);`

splitbrain commented on syntax.php:
```diff
@@ -72,11 +72,11 @@ function _link($target) {
         // '^(http://)?[-_[:alnum:]]+[-_.[:alnum:]]*\.[a-z]{2}'
         // '(/[-_./[:alnum:]&%?=#]*)?';
-        if (ereg($url, $target))
+        if (preg_match($url, $target))
```
preg and ereg have slightly different syntax. the expression needs adjustment

dunron replied:
Works when delimiting the static:

`        static $url = '#^https?://#';`

splitbrain reviewed (changes requested):
Just reading the code I doubt this works. Can you adjust the code and double check that it works as intended? (bonus points for unit tests)

dunron committed:
Update syntax.php

PHP 7 requires delimiters around the regex. Explode is more elegant than preg_split. Patches have been unit tested (URL and internal link).

dunron commented:
Could you please review the changes that I offered on 23 January? There may be more people running into this incompatibility issue. Thank you.

splitbrain commented:
Your commit mentions unit tests, but they are not included!?

dunron commented:
Indeed, I "unit tested" the patches. Honest, cross my heart and hope to die. Not sure what else you could be expecting from me. :-o

splitbrain commented:
I was actually asking for automated tests. Never mind.
